android/app/src/main/java/org/electroncash/electroncash3/NewWallet.kt

```diff
@@ -45,26 +56,40 @@ class NewWalletDialog1 : AlertDialogFragment() {
                     putString("password", password)
                 }
 
-                val walletType = spnType.selectedItemId.toInt()
-                if (walletType in listOf(R.id.menuCreateSeed, R.id.menuRestoreSeed)) {
-                    nextDialog = NewWalletSeedDialog()
-                    val seed = if (walletType == R.id.menuCreateSeed)
-                                   daemonModel.commands.callAttr("make_seed").toString()
-                               else null
-                    arguments.putString("seed", seed)
-                } else if (walletType == R.id.menuImport) {
-                    nextDialog = NewWalletImportDialog()
-                } else if (walletType == R.id.menuImportMaster) {
-                    nextDialog = NewWalletImportMasterDialog()
-                } else {
-                    throw Exception("Unknown item: ${spnType.selectedItem}")
+                when (spnWalletType.selectedItemId.toInt()) {
+                    R.id.menuStandardWallet -> {
+                        nextDialog = KeystoreDialog()
+                        showDialog(this, nextDialog.apply { setArguments(arguments) })
+                    }
+                    R.id.menuMultisigWallet -> {
+                        nextDialog = CosignerDialog()
+                        showDialog(this, nextDialog.apply { setArguments(arguments) })
+                    }
+                    else -> {
+                        throw Exception("Unknown item: ${spnWalletType.selectedItem}")
+                    }
                 }
-                showDialog(this, nextDialog.apply { setArguments(arguments) })
+
             } catch (e: ToastException) { e.show() }
         }
     }
+
+    override fun onDismiss(dialog: DialogInterface) {
+        super.onDismiss(dialog)
+        keystores.clear()
+    }
 }
 
+fun closeDialogs(targetFragment: Fragment) {
+
+    if (targetFragment.targetFragment == null) {
+        (targetFragment as DialogFragment).dismiss()
+        return
+    }
+
+    closeDialogs(targetFragment.targetFragment!!)
+    (targetFragment as DialogFragment).dismiss()
```

Comment:
Can't this be simplified to the following?

    (targetFragment as DialogFragment).dismiss()
    if (targetFragment.targetFragment != null) {
        closeDialogs(targetFragment.targetFragment!!)
    }